clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -7380,7 +7380,11 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
                    false))
     CmdArgs.push_back("-fmodules-debuginfo");
 
-  if (!CLANG_ENABLE_OPAQUE_POINTERS_INTERNAL)
+  // The output of this compilation will be the input to mlir-translate, which
+  // only works with opaque pointers.
+  if (Args.hasArg(options::OPT_fsycl_raise_host))
+    CmdArgs.push_back("-opaque-pointers");
```

Comment:
If CLANG_ENABLE_OPAQUE_POINTERS_INTERNAL is false and OPT_fsycl_raise_host is true can we emit a warning stating that the new option (OPT_fsycl_raise_host) cannot be used rather than silently ignore CLANG_ENABLE_OPAQUE_POINTERS_INTERNAL

Reply:
Fixed already :+1: